src/components/ThemeProvider/README.md

```diff
@@ -0,0 +1,230 @@
+      primaryFooterAction={{content: 'Add tracking number'}}
+    >
+      <Card.Section title="Items">
+        <List>
+          <List.Item>1 × Oasis Glass, 4-Pack</List.Item>
+          <List.Item>1 × Anubis Cup, 2-Pack</List.Item>
+        </List>
+      </Card.Section>
+    </Card>
+    <ThemeProvider theme={{colorScheme: 'dark'}}>
+      <Card
+        title="Shipment 1234"
+        secondaryFooterActions={[{content: 'Edit shipment'}]}
+        primaryFooterAction={{content: 'Add tracking number'}}
+      >
+        <Card.Section title="Items">
+          <List>
+            <List.Item>1 × Oasis Glass, 4-Pack</List.Item>
+            <List.Item>1 × Anubis Cup, 2-Pack</List.Item>
+          </List>
+        </Card.Section>
+      </Card>
+    </ThemeProvider>
+  </TextContainer>
+</AppProvider>
+```
+
+### Theme provider with colors nested within an app provider
+
+A theme provider can be nested within the theme provider rendered by the app provider in order to override colors at a local level.
+
+```jsx
+<AppProvider i18n={{}} features={{newDesignLanguage: true}}>
+  <Card>
+    <Card.Section>
+      <Stack alignment="center">
+        <Stack.Item fill>
+          This setting is <TextStyle variation="strong">disabled</TextStyle>.
+        </Stack.Item>
+        <Stack.Item>
+          <Button primary>Enable</Button>
+        </Stack.Item>
+      </Stack>
+    </Card.Section>
+    <Card.Section>
+      <Stack alignment="center">
+        <Stack.Item fill>
+          This setting is <TextStyle variation="strong">disabled</TextStyle>.
+        </Stack.Item>
+        <Stack.Item>
+          <ThemeProvider theme={{colors: {primary: '#3b5998'}}}>
+            <Button primary>Enable</Button>
+          </ThemeProvider>
+        </Stack.Item>
+      </Stack>
+    </Card.Section>
+  </Card>
+</AppProvider>
+```
+
+---
+
+## Consuming theme colors in a component
+
+The theme provider component uses [CSS custom properties](https://developer.mozilla.org/en-US/docs/Web/CSS/--*) to share color values with components. For a full list of available CSS custom properties, see the [Polaris tokens docs](link/to/polaris/tokens/docs).
+
+We apply the current design language value to the fallback value of the `var()` function in order to gracefully fall back when `{newDesignLanguage: false}`, as the theme CSS custom properties will not be defined.
+
+```scss
+.Card {
+  background-color: var(--p-surface, color('white'));
+  box-shadow: var(--p-card-shadow, shadow());
+  border-radius: var(--p-border-radius-wide, border-radius());
+}
+```
+
+### Avoid using `linear-gradient()` as a `var()` fallback
+
+Using the `linear-gradient()` function as a fallback to the `var()` function causes infinite re-paints in Safari, and should be avoided. As an alternative, use the value provided to `newDesignLanguage` to create a class name for your component that's only present when that value is `true`.
+
+```scss
+.Button {
+  background-color: linear-gradient(
```

Comment:
`linear-gradient` should be added to `background-image` as oppose to `background-color`